install.sh

```diff
@@ -49,16 +52,68 @@ else
 	fi
 fi
 
+
+allBuildArguments=("$@")
+for ((index=0; index < ${#allBuildArguments[@]}; index++)); do
+	if [ ${allBuildArguments[index]} = '-r' ]; then
+		reposArg=${allBuildArguments[index+1]}
+		if [ "$reposArg" = "" ]; then
+			echo "Missing required argument for -r"
+			exit 1 # exit script with error
+		fi
+		# parse the r option values
+		IFS=',' read -r -a repos <<< "$reposArg"
+	fi
+done
+
 if [ -d ../operator ]; then 
 
 	# pluck image tag off operator image 
 	tag=$(cat ../operator/kappnav.yaml | grep operator: | awk '{ split($0,p,":"); print p[3] }')
-
+	
 	echo Install kappnav to kubeenv $kubeenv
 	kubectl create namespace kappnav 
 
-	cat ../operator/kappnav.yaml | sed "s|kubeEnv: okd|kubeEnv: $kubeenv|" | sed "s|repository: kappnav/|repository: $org/kappnav-|" | sed "s|tag: $tag|tag: latest|" | sed "s|image: kappnav/operator:$tag|image: $org/kappnav-operator:latest|" | kubectl create -f - -n kappnav 
+	if [[ x$reposArg == 'x' || x$reposArg == "" ]]; then
+		cat ../operator/kappnav.yaml | sed "s|kubeEnv: okd|kubeEnv: $kubeenv|" | sed "s|repository: kappnav/|repository: $org/kappnav-|" | sed "s|tag: $tag|tag: latest|" | sed "s|image: kappnav/operator:$tag|image: $org/kappnav-operator:latest|" | kubectl create -f - -n kappnav 
+	else
+		org=$DOCKER_USER
+		echo $DOCKER_PWD | docker login docker.io -u $DOCKER_USER --password-stdin
+		. ./projectList.sh
+		projs=$BUILD_PROJECTS
+		
+		cat ../operator/kappnav.yaml | \
+			sed "s|kubeEnv: okd|kubeEnv: $kubeenv|" | \
+			sed "s|tag: $tag|tag: dev|" | \
+			sed "s|image: kappnav/operator:$tag|image: kappnav/operator:dev|" \
+			> temp-kappnav.yaml
+		for p in $projs; do
+			for repo in "${repos[@]}"; do
+				if [ x$repo == x$p ]; then
+					if [ x$repo == x"operator" ]; then
+						sed "s|image: kappnav/operator:dev|image: $DOCKER_USER/kappnav-operator:latest|" temp-kappnav.yaml > temp-kappnav-new.yaml
+					else
+						if [ x$repo == x"inventory" ]; then
+							repo="inv"
+						fi
+						r="repository: kappnav/"$repo
+						# get the line number of the repo that the tag need to be updated
+						ln=`grep -n "$r" ../operator/kappnav.yaml | awk -F: '{print $1}'`
+						newln=$(($ln+1))
+						cat temp-kappnav.yaml | \
+						sed "$ln s|repository: kappnav/|repository: $DOCKER_USER/kappnav-|" | \
+						sed "$newln s|tag: dev|tag: latest|" > temp-kappnav-new.yaml
+					fi
+					cat temp-kappnav-new.yaml > temp-kappnav.yaml
+				fi
+			done
+		done
+		kubectl create -f temp-kappnav.yaml -n kappnav
+	fi
 else
 	echo Cannot install: file ../operator/kappnav.yaml not found. 
 	exit 1
 fi
+
+# remove temp file created by this script
+rm -fr temp-kappnav*.yaml
```

Comment:
Why do we need the recursive option here?

Reply:
Can please tell me te line number of what recursive option you meant?

Reply:
119  -r

Reply:
sorry, I thought fr is for force. I will remove the r. thx Roger.